src/mmap_unix.rs

```diff
@@ -51,6 +51,73 @@ pub enum Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) struct Mmap {
+    addr: *mut u8,
+    size: usize,
+    owned: bool,
+}
+
+impl Mmap {
+    pub fn new(size: usize, prot: i32, flags: i32, fd: i32, f_offset: u64) -> Result<Self> {
+        let addr =
+            // SAFETY: This is safe because we're not allowing MAP_FIXED, and invalid parameters
+            // cannot break Rust safety guarantees (things may change if we're mapping /dev/mem or
+            // some wacky file).
+            unsafe { libc::mmap(null_mut(), size, prot, flags, fd, f_offset as libc::off_t) };
```

Comment:
Do we need some mechanism to prevent mapping the same region twice? Especially with subslicing going on it seems to be relatively easy to map the same region twice? Does not seem to be a new issue though... The code just moved.

Reply:
This is not an issue with the current code at least, it can happen only after Xen bits come in. And there too, I am not sure if we will end up having that optimization to avoid code complexity.